content/tutorials/foundry-era-test-node/10.index.md

```diff
@@ -0,0 +1,290 @@
+---
+title: Interacting with Uniswap on a Fork of ZKsync Era Using `era_test_node` and Foundry
+description: A step-by-step guide to set up a fork of ZKsync Era mainnet and interact with the Uniswap contract to perform token swaps using Foundry.
+---
+
+This tutorial shows you how to set up a local ZKsync Era environment using `era_test_node`,
+leverage Foundry's `cast` tool to interact with contracts, and perform token swaps on Uniswap.
+You'll learn how to:
+
+1. Set up your development environment.
+2. Streamline your workflow with environment variables.
+3. Check balances using `cast call`.
+4. Deposit ETH for WETH.
+5. Transfer WETH to another address.
+6. Swap WETH for an ERC20 token on Uniswap.
+
+Let's get started!
+
+---
+
+## Prerequisites
+
+- **Foundry**: Ethereum development toolkit. [Installation Guide](https://book.getfoundry.sh/getting-started/installation)
+- **era_test_node**: In-memory node to fork ZKsync Era mainnet. [Installation Guide](https://docs.zksync.io/build/test-and-debug/in-memory-node#install-and-set-up-era_test_node)
+- **Online Converters**:
+  - [ETH Converter](https://eth-converter.com/) for converting ETH values.
+  - [Hex to Decimal Converter](https://www.rapidtables.com/convert/number/hex-to-decimal.html) for number conversions.
+- **ZKsync Era Explorer**:
+  - [ZKsync Era Explorer](https://era.zksync.io/) to view contracts.
+  - [Tokens List](https://era.zksync.io/tokens) to find token addresses.
+- **Uniswap Pool Explorer**:
+  - [Uniswap Pools on ZKsync Era](https://app.uniswap.org/explore/pools/zksync) to find pools for swapping.
+- **Price Conversion Tool**:
+  - [Kraken Convert](https://www.kraken.com/convert) to get current exchange rates.
+
+::callout{icon="i-heroicons-exclamation-triangle"}
+**Security Tip**: Never commit your private key to version control or share it publicly.
+Ensure your `.env` file is excluded from source control. `.env` should be used for
+development purposes only, see the [foundry keystore](https://docs.zksync.io/build/tooling/foundry/installation#private-key-setup-with-foundry-keystore)
+section of the docs for a more secure way to manage your private keys.
+
+::
+
+---
+
+## 1. Setting Up the Environment
+
+### Fork ZKsync Era Mainnet
+
+Start `era_test_node` and fork the ZKsync Era mainnet:
+
+```bash
+era_test_node fork mainnet
+```
+
+For debug mode (provides more transaction details):
+
+```bash
+era_test_node -d fork mainnet
```

Comment:
This doesn't work for me, it says there is no `-d` flag. What version of `era_test_node` do you have?

Reply:
era_test_node 0.1.0-alpha.27

Tested and it worked for me.

Reply:
Added a note about the version too